From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Sun, 2 Aug 2026 23:01:07 +0200
Subject: [PATCH 090/117] Bluetooth: btbcm: Mark BCM4355C0 default address
 invalid

BCM4355C0 controllers can report a chip-default address before board
configuration. Treat it as invalid so a unique address can be supplied.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: dea7a97c9a6f8d4576f74c4e0871e2de585083a3
---
 drivers/bluetooth/btbcm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/bluetooth/btbcm.c b/drivers/bluetooth/btbcm.c
index 31a81da91..fc4d33cf4 100644
--- a/drivers/bluetooth/btbcm.c
+++ b/drivers/bluetooth/btbcm.c
@@ -30,6 +30,7 @@
 #define BDADDR_BCM4330B1 (&(bdaddr_t) {{0x00, 0x00, 0x00, 0xb1, 0x30, 0x43}})
 #define BDADDR_BCM4334B0 (&(bdaddr_t) {{0x00, 0x00, 0x00, 0xb0, 0x34, 0x43}})
 #define BDADDR_BCM4345C5 (&(bdaddr_t) {{0xac, 0x1f, 0x00, 0xc5, 0x45, 0x43}})
+#define BDADDR_BCM4355C0 (&(bdaddr_t) {{0xac, 0x1f, 0x00, 0xc0, 0x55, 0x43}})
 #define BDADDR_BCM4355C1 (&(bdaddr_t) {{0x93, 0x76, 0x00, 0xc1, 0x55, 0x43}})
 #define BDADDR_BCM43341B (&(bdaddr_t) {{0xac, 0x1f, 0x00, 0x1b, 0x34, 0x43}})
 
@@ -122,6 +123,9 @@ int btbcm_check_bdaddr(struct hci_dev *hdev)
 	 * The address AA:AA:AA:AA:AA:AA indicates a BCM43430A1 controller
 	 * with no configured address.
 	 *
+	 * The address 43:55:C0:00:1F:AC indicates a BCM4355C0 controller
+	 * with no configured address.
+	 *
 	 * The address 43:55:C1:00:76:93 indicates a BCM4355C1 controller
 	 * with no configured address.
 	 */
@@ -132,6 +136,7 @@ int btbcm_check_bdaddr(struct hci_dev *hdev)
 	    !bacmp(&bda->bdaddr, BDADDR_BCM4330B1) ||
 	    !bacmp(&bda->bdaddr, BDADDR_BCM4334B0) ||
 	    !bacmp(&bda->bdaddr, BDADDR_BCM4345C5) ||
+	    !bacmp(&bda->bdaddr, BDADDR_BCM4355C0) ||
 	    !bacmp(&bda->bdaddr, BDADDR_BCM4355C1) ||
 	    !bacmp(&bda->bdaddr, BDADDR_BCM43430A0) ||
 	    !bacmp(&bda->bdaddr, BDADDR_BCM43430A1) ||
